openedx_events/event_bus/avro/tests/test_deserializer.py

```diff
@@ -175,3 +176,54 @@ def test_deserialization_of_nested_optional_fields(self):
         nested_field = data_dict["data"].field_0
         self.assertIsInstance(nested_field, SimpleAttrsWithDefaults)
         self.assertEqual(nested_field, SimpleAttrsWithDefaults())
+
+    def test_deserialization_of_list_with_annotation(self):
+        """
+        Check that deserialization works as expected when list data is annotated.
+        """
+        LIST_SIGNAL = create_simple_signal({"list_input": List[int]})
+        initial_dict = {"list_input": [1, 3]}
+        deserializer = AvroSignalDeserializer(LIST_SIGNAL)
+        event_data = deserializer.from_dict(initial_dict)
+        expected_event_data = [1, 3]
+        test_data = event_data["list_input"]
+        self.assertIsInstance(test_data, list)
+        self.assertEqual(test_data, expected_event_data)
+
+    def test_deserialization_of_list_without_annotation(self):
+        """
+        Check that deserialization raises error when list data is not annotated.
+        """
+        # create dummy signal to test deserializer
+        SIGNAL = create_simple_signal({"list_input": List[int]})
```

Comment:
Can we clarify this comment? I had to go further down to understand why we needed two Signals here

Reply:
Ahh yes, I missed to explain this part in comments. Updated, let me know if it is clear now.